sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryExceptionDataTests.cs

```diff
@@ -0,0 +1,198 @@
+        {
+            Assert.Throws<ArgumentNullException>(() => new TelemetryExceptionDetails(null, "Exception Message", null));
+        }
+
+        [Fact]
+        public void ParsedStackIsEmptyForExceptionWithoutStack()
+        {
+            var exception = new ArgumentNullException();
+
+            var exceptionDetails = new TelemetryExceptionDetails(exception, exception.Message, null);
+
+            Assert.Equal(0, exceptionDetails.ParsedStack.Count);
+
+            // hasFullStack defaults to true.
+            Assert.True(exceptionDetails.HasFullStack);
+        }
+
+        [Fact]
+        public void AllStackFramesAreConvertedIfSizeOfParsedStackIsLessOrEqualToMaximum()
+        {
+            var exception = CreateException(42);
+
+            var exceptionDetails = new TelemetryExceptionDetails(exception, exception.Message, null);
+            Assert.Equal(43, exceptionDetails.ParsedStack.Count);
+            Assert.True(exceptionDetails.HasFullStack);
+        }
+
+        [Fact]
+        public void FirstAndLastStackPointsAreCollectedForLongStack()
+        {
+            var exception = CreateException(300);
+
+            var exceptionDetails = new TelemetryExceptionDetails(exception, exception.Message, null);
+
+            Assert.False(exceptionDetails.HasFullStack);
+            Assert.True(exceptionDetails.ParsedStack.Count < 300);
+
+            // We should keep top of stack, and end of stack hence CreateException function should be present
+            Assert.Equal(typeof(TelemetryExceptionDataTests).FullName + "." + nameof(FunctionWithException), exceptionDetails.ParsedStack[0].Method);
+            Assert.Equal(typeof(TelemetryExceptionDataTests).FullName + "." + nameof(CreateException), exceptionDetails.ParsedStack[exceptionDetails.ParsedStack.Count - 1].Method);
+        }
+
+        [Fact]
+        public void TestNullMethodInfoInStack()
+        {
+            var frameMock = new Mock<System.Diagnostics.StackFrame>(null, 0, 0);
+            frameMock.Setup(x => x.GetMethod()).Returns((MethodBase)null);
+
+            Models.StackFrame stackFrame = null;
+
+            stackFrame = TelemetryExceptionDetails.GetStackFrame(frameMock.Object, 0);
+
+            Assert.Equal("unknown", stackFrame.Assembly);
+            Assert.Null(stackFrame.FileName);
+            Assert.Equal("unknown", stackFrame.Method);
+            Assert.Null(stackFrame.Line);
+        }
+
+        [Fact]
+        public void CheckThatFileNameAndLineAreCorrectIfAvailable()
+        {
+            var exception = CreateException(1);
+
+            StackTrace st = new StackTrace(exception, true);
+            var frame = st.GetFrame(0);
+            var line = frame.GetFileLineNumber();
+            var fileName = frame.GetFileName();
+
+            var exceptionDetails = new TelemetryExceptionDetails(exception, exception.Message, null);
+            var stack = exceptionDetails.ParsedStack;
+
```

Comment:
nit: Leave a comment why line=0 is needed.

Reply:
line = 0 was the case in live unit test for some reason. Its not failing otherwise. Removed it.